tests/integ/sagemaker/workflow/test_workflow.py

```diff
@@ -1222,17 +1225,103 @@ def test_get_last_execution(sagemaker_session, role, pipeline_name, region_name)
 
         output = pipeline.get_last_execution()
         output_successful = pipeline.get_last_execution(successful=True)
+        list_response = pipeline.list_executions()
+        output_list = list_response["PipelineExecutionList"]
 
-        assert output.arn == execution_fail.arn
-        assert output.pipeline == pipeline
+        equal_pipeline_executions(output, execution_fail)
         assert output.describe()["PipelineExecutionStatus"] == "Failed"
 
-        assert output_successful.arn == execution_success.arn
-        assert output_successful.pipeline == pipeline
+        equal_pipeline_executions(output_successful, execution_success)
         assert output_successful.describe()["PipelineExecutionStatus"] == "Succeeded"
 
+        equal_pipeline_executions(output_list[0], execution_fail)
+        assert output_list[0].describe()["PipelineExecutionStatus"] == "Failed"
+        equal_pipeline_executions(output_list[1], output_successful)
+        assert output_list[1].describe()["PipelineExecutionStatus"] == "Succeeded"
+        assert "NextToken" not in list_response
+
     finally:
         try:
             pipeline.delete()
         except Exception:
             pass
+            
```

Comment:
Can we also test the case when NextToken is not None? As list executions and list pipelines are quite similar, we can test pagination for list executions only.
We can set the page size to be a small value like 2 and have 3 executions.

This would require us to expose the MaxResults (have sent model link to you in slack)